source/NVDAObjects/UIA/__init__.py

```diff
@@ -1832,6 +1832,25 @@ def event_UIA_notification(self, notificationKind=None, notificationProcessing=U
 				speech.cancelSpeech()
 			ui.message(displayString)
 
+	def event_UIA_dragComplete(self):
+		"""
+		Raised by an element when a drag operation on it is completed.
+		A prominent example is rearranging Start screen tiles on Windows 8 and later.
+		"""
+		# #12271: speech does not report new location but braille does.
+		self.event_gainFocus()
+
+	def event_UIA_droptTargetDropped(self):
+		"""
+		Raised by a containing element such as a list when an element is dragged over another element.
+		A prominent example is rearranging quick actions in Windows 10 Action Center.
+		"""
+		# #12271: unlike drag complete event, it is a different element that raises this event.
+		# Specifically, it is the containing element that raises this event.
+		# but NVDA does record the newly dragged element as focused object.

```

Comment:
How is the newly dragged element focused by NVDA?

Reply:
I think the following traceback may explain this a bit (debug log with MSAA and UIA flags turned on, along with Windows 10 App Essentials enabled; scenario: an item in Action Center was rearranged):

IO - inputCore.InputManager.executeGesture (01:23:37.906) - winInputHook (1140):
Input: kb(desktop):alt+shift+leftArrow
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (01:23:37.927) - MainThread (7948):
Hook received winEvent: EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 176, childID 0, process 2844 (shellexperiencehost), thread 10584
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (01:23:37.928) - MainThread (7948):
Adding winEvent to limiter: EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 176, childID 0, process 2844 (shellexperiencehost), thread 10584
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (01:23:37.941) - MainThread (7948):
Hook received winEvent: EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 177, childID 0, process 2844 (shellexperiencehost), thread 10584
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (01:23:37.941) - MainThread (7948):
Adding winEvent to limiter: EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 177, childID 0, process 2844 (shellexperiencehost), thread 10584
DEBUG - IAccessibleHandler.orderedWinEventLimiter.OrderedWinEventLimiter.flushEvents (01:23:37.946) - MainThread (7948):
Emitting winEvent EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 176, childID 0, process 2844 (shellexperiencehost), thread 10584
DEBUG - IAccessibleHandler.orderedWinEventLimiter.OrderedWinEventLimiter.flushEvents (01:23:37.946) - MainThread (7948):
Emitting winEvent EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 177, childID 0, process 2844 (shellexperiencehost), thread 10584
DEBUG - IAccessibleHandler.processFocusWinEvent (01:23:37.949) - MainThread (7948):
Processing focus winEvent: window 3281400 (Windows.UI.Core.CoreWindow), objectID 177, childID 0, process 2844 (shellexperiencehost), force False
DEBUG - IAccessibleHandler.winEventToNVDAEvent (01:23:37.949) - MainThread (7948):
Creating NVDA event from winEvent: EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 177, childID 0, process 2844 (shellexperiencehost), use cache False
DEBUG - IAccessibleHandler.winEventToNVDAEvent (01:23:37.950) - MainThread (7948):
winEvent mapped to NVDA event: gainFocus
DEBUG - IAccessibleHandler.winEventToNVDAEvent (01:23:37.950) - MainThread (7948):
Native UIA window. Dropping winEvent EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 177, childID 0, process 2844 (shellexperiencehost)
DEBUG - IAccessibleHandler.processFocusWinEvent (01:23:37.951) - MainThread (7948):
Processing focus winEvent: window 3281400 (Windows.UI.Core.CoreWindow), objectID 176, childID 0, process 2844 (shellexperiencehost), force False
DEBUG - IAccessibleHandler.winEventToNVDAEvent (01:23:37.951) - MainThread (7948):
Creating NVDA event from winEvent: EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 176, childID 0, process 2844 (shellexperiencehost), use cache False
DEBUG - IAccessibleHandler.winEventToNVDAEvent (01:23:37.951) - MainThread (7948):
winEvent mapped to NVDA event: gainFocus
DEBUG - IAccessibleHandler.winEventToNVDAEvent (01:23:37.951) - MainThread (7948):
Native UIA window. Dropping winEvent EVENT_OBJECT_FOCUS, window 3281400 (Windows.UI.Core.CoreWindow), objectID 176, childID 0, process 2844 (shellexperiencehost)
DEBUG - _UIAHandler.UIAHandler.IUIAutomationFocusChangedEventHandler_HandleFocusChangedEvent (01:23:37.982) - Dummy-3 (12344):
HandleFocusChangedEvent: Ignoring because no object or ignored by object itself
DEBUGWARNING - _UIAHandler.UIAHandler.IUIAutomationFocusChangedEventHandler_HandleFocusChangedEvent (01:23:37.985) - Dummy-3 (12344):
HandleFocusChangedEvent: Ignoring duplicate focus event
DEBUG - external:globalPlugins.wintenObjs.GlobalPlugin.uiaDebugLogging (01:23:38.021) - MainThread (7948):
W10: UIA object: <NVDAObjects.UIA.UIA object at 0x072874D0>, name: , event: dropTargetDropped, app module: <'shellexperiencehost' (appName 'shellexperiencehost', process ID 2844) at address 6cb1df0>, Automation Id: ScrollWrapper, class name: ScrollViewer

NOte that this was with selective UIA event registration enabled.

Thanks.